### DISCUSSION TITLE
Overlay for loading

### DISCUSSION THREAD

**Discussion by cubecleveland** [table-library/react-table-library#21] [Q&A] [answered]
Started using this lib and it looks great.
Its one of the only one i could find that is geared toward server rendering...

Is there any native way to show an ovelray when the table is loading ?

**Comment by rwieruch** [comment]
There is nothing native for it. However, you can do this yourself! I give you some hints:

Take the Server/Search/with Callback from the [Storybook](https://react-table-library.com/?path=/docs/server-search--with-callback) for example. In the `onSearchChange` function's body, you could use a `isLoading` state. Toggle it to true before the `doGet ` call and to false after the `doGet` call. Then use this state to render a loading indicator. 

If you want to have a loading indicator which is displayed as overlay, you just need to create such overlay component ([at the end, it's just some CSS](https://stackoverflow.com/questions/6091253/overlay-with-spinner/20013732)) next to the table component.

However, I try to find the time this week to make an example for it!

**Comment by rwieruch** [comment, marked as answer]
Story: https://react-table-library.com/?path=/story/server-recipes-overlay--base 
Source Code: https://github.com/table-library/react-table-library/blob/master/.storybook/stories/Server/Recipes/overlay.story.js
